### DISCUSSION TITLE
Standard way of saving Essentia data to media file tags

### DISCUSSION THREAD

**Discussion by JOJ0** [MTG/essentia#1239] [Q&A] [answered]
Hi,
I was wondering if there is a standard set of names that should be used when Essentia analysation data is saved to actual media files, eg. ID3 tags?

The reason why I'm asking is that I was about to bring a feature into the AcousticBrainz plugin for the music tagger https://github.com/beetbox/beets that does write a subset of the AB-provided data directly into the files tags. We were in contact with @wargreen who wrote the AcousticBrainz plugin for the MusicBrainz Picard tagger.

Picard uses file tags looking like these:
```
TXXX=ab:hi:danceability:danceable=0.000000000000
TXXX=ab:hi:genre_rosamerica=rhy
TXXX=ab:hi:ismir04_rhythm=ChaChaCha
TXXX=ab:hi:mood_acoustic:acoustic=0.081802986562
TXXX=ab:hi:mood_aggressive:aggressive=0.000000000000
TXXX=ab:lo:tonal:chords_key=A#
TXXX=ab:lo:tonal:chords_number_rate=0.002038320526
TXXX=ab:lo:tonal:chords_scale=minor
TXXX=ab:lo:tonal:key_strength=0.654201686382
```

 and we were about to implement it equally into the beets plugin (https://beets.readthedocs.io/en/stable/plugins/acousticbrainz.html).

But now, since AcousticBrainz is going to be discontinued, the filetag names consting of prefixes AB:HI and AB:LO don't seem to be suitable any more. So the idea is to bring in Essentia data directly into file tags. A plugin for beets that uses the Essentia binary extractors is existing already (https://github.com/adamjakab/BeetsPluginXtractor#xtractor-beets-plugin).

I just wanted to come here before moving on and ask if you are aware of other software that is writing this data to tags already and if there is a common "namespace prefix" or something that we should follow. If there is none I'd like to open the discussion on tag names. Some basic ideas on my end would be to simply exchange the ab:xx: in the naming scheme Picard was introducing with essentia: (or ess: since the names are quite long already), so then get something like

```TXXX=essentia:genre_rosamerica=rhy```

or

```TXXX=ess:genre_rosamerica=rhy```

Thanks in advance!

Update: To be more specific, what I'm talking about is using the binary extractors `essentia_streaming_extractor_music` since they are easy to install for everyone. AcousticBrainz used version 2.1 beta2 it seems: http://ftp.acousticbrainz.org/pub/acousticbrainz/essentia-extractor-v2.1_beta2-linux-x86_64.tar.gz

which is available from your website, e.g here: https://essentia.upf.edu/extractors/essentia-extractors-v2.1_beta2-linux-x86_64.tar.gz as documented here: https://github.com/MTG/essentia/blob/master/doc/sphinxdoc/extractors_out_of_box.rst

**Comment by JOJ0** [comment]
@dbogdanov and @palonso seems you are maintaining the project. Any chance you could help me out with your thoughts on the topic? Sorry for bothering and thanks in advance! :-)))

**Comment by dbogdanov** [comment, marked as answer]
Hi, @JOJ0. It is great to see you are working on this. I am using beets and considering to propose a new Essentia analysis plugin too. 

I like the idea of a full name for namespace prefix: `essentia:` instead of  `ess:`  - this is less ambiguous. 

Given that we have an updated version of Music Extractor (`essentia_streaming_extractor_music`, previously used for AcousticBrainz) and various new machine learning classifiers we created, I propose to have a way to distinguish the AB-compatible analysis features further: `essentia:ab:`.
